nion/swift/ExportDialog.py

```diff
@@ -221,56 +223,163 @@ def cancel(self) -> bool:
+                new_width = new_height * self.aspect_ratio
+                if self.unit_model == UnitType.PIXELS:
+                    new_width = round(new_width)
+                self.width_value_line_edit.text = str(new_width)
+                self.width_value_line_edit.background_color = "#lightgrey"
+                self.height_value_line_edit.background_color = "white"
+        except ValueError:
+            pass  # Handle the case where the input is not a valid integer
+
+    def unit_changed(self, widget: UserInterface.ComboBoxWidget, current_index: int) -> None:
+        new_unit_type = UnitType(current_index)
+        self.size = self.convert_to_pixels(
+            Geometry.FloatSize(height=float(self.height_value_line_edit.text or 0),
+                               width=float(self.width_value_line_edit.text or 0)),
+            self.previous_unit_index
+        )
+        new_width, new_height = self.convert_from_pixels(self.size, new_unit_type)
+        if new_unit_type == UnitType.PIXELS:
+            new_width = round(new_width)
+            new_height = round(new_height)
+        else:
+            new_width = round(new_width, 6)
+            new_height = round(new_height, 6)
+        self.width_value_line_edit.text = str(new_width)
+        self.height_value_line_edit.text = str(new_height)
+        self.previous_unit_index = new_unit_type
+
+    def convert_to_pixels(self, size: Geometry.FloatSize, unit_type: UnitType) -> Geometry.IntSize:
+        if size.height is None or size.width is None:
+            return Geometry.IntSize(0, 0)
+        else:
+            height = float(size.height)
+            width = float(size.width)
+        if unit_type == UnitType.INCHES:
+            return Geometry.IntSize(height=int(height * PIXELS_PER_INCH), width=int(width * PIXELS_PER_INCH))
+        elif unit_type == UnitType.CENTIMETERS:
+            return Geometry.IntSize(height=int(height * PIXELS_PER_INCH / CM_PER_INCH),
+                                    width=int(width * PIXELS_PER_INCH / CM_PER_INCH))
+        return Geometry.IntSize(height=int(round(height)), width=int(round(width)))  # Pixels
+
+    def convert_from_pixels(self, size: Geometry.IntSize, unit_type: UnitType) -> Geometry.FloatSize:
+        if size.height is None or size.width is None:
+            return Geometry.FloatSize(0.0, 0.0)
+        else:
+            height_px = float(size.height)
+            width_px = float(size.width)
+        if unit_type == UnitType.INCHES:
+            return Geometry.FloatSize(height=height_px / PIXELS_PER_INCH, width=width_px / PIXELS_PER_INCH)
+        elif unit_type == UnitType.CENTIMETERS:
+            return Geometry.FloatSize(height=height_px * CM_PER_INCH / PIXELS_PER_INCH,
+                                      width=width_px * CM_PER_INCH / PIXELS_PER_INCH)
+        return Geometry.FloatSize(width=width_px, height=height_px)  # Pixels
+
     def close(self) -> None:
         pass
 
-
```

Comment:
There should be exactly two blank lines before each top-level class. Please add this one back.